### PR TITLE
[WIP] akka persistence query for SQL journals

### PR DESCRIPTION
**Pull request by Horusiath**
**WARNING: Work in progress**

This PR includes new version of sql-based persistence plugins working with `Akka.Persistence.Query` - akka streams plugin for the persistence queries read-side. It also has a working example in form of new version of `Akka.Persistence.Sqlite` plugin - at the moment all tests are passing.

Note: all sql based plugins can make use of `Akka.Persistence.Query.Sql` - it will work with all of them. Only write-side needs to be provided separately.

### CODE REVIEW

**Comment by alexvaluyskiy** [comment]
Maybe you could incorporate this PR into yours?

**Comment by alexvaluyskiy** [comment]
Seems to be, you have already implemented the journal test, but still miss a snapshot test.

You have a default sql in Sql.Common, do you thinkit is a good idea?

**Comment by Horusiath** [comment]
Thx @alexvaluyskiy . I'll take your changes, once I get a second to upgrade this PR ;)

File: src/contrib/persistence/Akka.Persistence.Query.Sql/SqlReadJournal.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    {
+        public static string Identifier = "akka.persistence.query.journal.sql";
+
+        /// <summary>
+        /// Returns a default query configuration for akka persistence SQLite-based journals and snapshot stores.
+        /// </summary>
+        /// <returns></returns>
+        public static Config DefaultConfiguration()
+        {
+            return ConfigurationFactory.FromResource<SqlReadJournal>("Akka.Persistence.Query.Sql.reference.conf");
```
Comment (alexvaluyskiy) [inline src/contrib/persistence/Akka.Persistence.Query.Sql/SqlReadJournal.cs]: Absent tests for default configuration

File: src/contrib/persistence/Akka.Persistence.Query.Sql/SqlReadJournal.cs [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        /// </para>
+        /// The returned event stream is unordered and you can expect different order for multiple
+        /// executions of the query.
+        /// <para>
+        /// The stream is not completed when it reaches the end of the currently used `persistenceIds`,
+        /// but it continues to push new `persistenceIds` when new persistent actors are created.
+        /// Corresponding query that is completed when it reaches the end of the currently
+        /// currently used `persistenceIds` is provided by <see cref="CurrentPersistenceIds"/>.
+        /// </para>
+        /// The LevelDB write journal is notifying the query side as soon as new `persistenceIds` are
```
Comment (alexvaluyskiy) [inline src/contrib/persistence/Akka.Persistence.Query.Sql/SqlReadJournal.cs]: You have LevelDB and Sqlite in XML docs. Maybe you should change all string to Sqlite?

**Pull request closed without merging**